core/src/main/java/org/apache/calcite/rel/metadata/RelMdRowCount.java

```diff
@@ -169,6 +170,12 @@ public Double getRowCount(Calc rel, RelMetadataQuery mq) {
     return limit < rowCount ? limit : rowCount;
   }
 
+  public @Nullable Double getRowCount(Sample rel, RelMetadataQuery mq) {
+    final Double rowCount = mq.getRowCount(rel.getInput());
```

Comment:
nitpick: I'd name this variable `inputRowCount` or something like that.
Also, as Julian mentions, perhaps a comment would be helpful.